pallets/pallet-asset-switch/src/switch.rs

```diff
@@ -51,8 +92,77 @@ pub enum SwitchPairStatus {
 	Running,
 }
 
+// Constructor impls
 impl<AccountId> SwitchPairInfo<AccountId> {
```

Comment:
Because I wanted to express the fact that the input is not checked for its constraints here. Implementing simply `From` does not give that guarantee, but quite the opposite. It supposed to be infallible and should never have any assumptions. That's why you don't have it for numeric type but you rather have checked, saturating, and overflowing variants.